src/storage/src/source/kafka.rs

```diff
@@ -558,9 +562,61 @@ fn create_kafka_config(
         kafka_config.set(k, v);
     }
 
+    // Fixup any secret values in the config
+    fixup_secrets(&mut kafka_config);
+
     kafka_config
 }
 
+/// Find secrets in the ClientConfig and transform them to values or canonical paths as appropriate
+/// The key strings to expect come from mz_sql::kafka_util::extract_config
+fn fixup_secrets(cfg: &mut ClientConfig) {
```

Comment:
We need to do something more robust than trying to re-sniff secret values out of the config map. We control this string map, so we're just causing ourselves this trouble by not retaining the information about what was a secret and what was not!

I recommend changing the value type of the `config_options` map stored in `KafkaSourceOptions` to be something like:

```rust
enum KafkaConfigValue {
    Secret(GlobalId),
    Literal(String),
}
```

Then you won't need to guess here about what's a secret and what's not.

Reply:
The original reason these needed to be here was to allow for compatibility with the existing `CREATE SOURCE` syntax when not using connectors and secrets. This code was also expecting that we would not allow the planner to read secrets (that was added to make schema registry connections in `mz_interchange` work) so they would come over either as correct values or stringified GlobalIds which would need to be read or canonicalized which is why is so defensive and assumes if it cannot be turned into a secret then it was already a valid value.